test/rest-router.test.js

```diff
@@ -0,0 +1,93 @@
+var assert = require('assert');
+var expect = require('chai').expect;
+var RestRouter = require('../lib/rest-router.js');
+var Trie = require('../lib/trie.js');
+var express = require('express');
+var supertest = require('supertest');
+var http = require('http');
+
+describe('RestRouter', function() {
+  // Test init
+  it('initializes with Trie datastructure', function(done) {
+    var router = new RestRouter();
+    expect(router).to.have.property('_trie');
+    expect(router._trie).to.be.instanceOf(Trie);
+    done();
+  });
+
+  // test handle
+  it('handles a request', function(done) {
+    var app = express();
+    var router = new RestRouter();
+    var route = { fullPath: '/example', verb: 'get' };
+    var resBody = 'Hello Wold!';
+    // this doesn't make sense to me that I have to call router.get(...)
+    // as well as have route.verb = get
+    // it doesn't work without it
+    router.get(route, function(req, res) {
+      res.end(resBody);
+    });
+    app.use(router);
+    console.log(router._trie);
+    supertest(app)
+      .get(route.fullPath)
+      .expect(200)
+      .expect(resBody)
+      .end(function(err, res) {
+        if (err) return done(err);
+        done();
+      });
+  });
+
+  // test handle param
+  it('handles a request with a parameter', function(done) {
+    var app = express();
+    var router = new RestRouter();
+    var route = { fullPath: '/example/:id', verb: 'get' };
+    var id = '1';
+    // this doesn't make sense to me that I have to call router.get(...)
+    // as well as have route.verb = get
+    // it doesn't work without it
+    router.get(route, function(req, res, next) {
+      console.log('hit');
+      res.end(id);
```

Comment:
In TDD, it's crucial to start with a failing unit test, to see the test fail and fail for the right (expected) reason. Without that, it's easy to write a test that always passed (like this one) or a test that fails for completely different reasons because it's based on wrong assumptions about how the code under test works.

Last but not least, seeing the test fail is the opportunity to verify that the failure message makes it clear what went wrong, what was expected and actual result.